server/services/TransferService.js

```diff
@@ -82,13 +84,91 @@ class TransferService {
         );
       }
 
+      const tokenArr = gottentokens.map((token) => token.id);
+
       let status;
       if (result.state === TransferEnums.STATE.completed) {
+        // the log should show up on both sender and receiver
+        await this._event.logEvent({
+          loggedInWalletId: walletSender.id,
+          type: 'transfer_completed',
+          payload: tokens
+            ? {
+                walletSender: walletSender.name,
+                walletReceiver: walletReceiver.name,
+                tokenTransferred: tokenArr,
+                claim,
+              }
+            : {
+                walletSender: walletSender.name,
+                walletReceiver: walletReceiver.name,
+                bundle: bundle.bundle_size,
+                claim,
+              },
+        });
+
+        // the log should show up on both sender and receiver
+        await this._event.logEvent({
+          loggedInWalletId: walletReceiver.id,
+          type: 'transfer_completed',
+          payload: tokens
+            ? {
+                walletSender: walletSender.name,
+                walletReceiver: walletReceiver.name,
+                tokenTransferred: tokenArr,
+                claim,
+              }
+            : {
+                walletSender: walletSender.name,
+                walletReceiver: walletReceiver.name,
+                bundle: bundle.bundle_size,
+                claim,
+              },
+        });
+
         status = 201;
       } else if (
         result.state === TransferEnums.STATE.pending ||
         result.state === TransferEnums.STATE.requested
       ) {
+        // the log should show up on both sender and receiver
+        await this._event.logEvent({
+          loggedInWalletId: walletSender.id,
+          type: 'transfer_requested',
```

Comment:
let's capture the current state since it can be either pending or requested

Reply:
added the state as transferState in the payload